src/gameLogic/generation/ActionGenerator.hpp

```diff
@@ -26,31 +27,33 @@ namespace spy::gameplay {
 
             static std::shared_ptr<BaseOperation> generateJanitorAction(const State &s);
 
-            static std::shared_ptr<BaseOperation> generateNPCAction(const State &s, const util::UUID &activeNPC);
+            static std::shared_ptr<BaseOperation>
+            generateNPCAction(const State &s, const util::UUID &activeNPC, const MatchConfig &config);
 
             static std::shared_ptr<BaseOperation>
-            generateRandomAction(const State &s, const util::UUID &activeCharacter);
+            generateRandomAction(const State &s, const util::UUID &activeCharacter, const MatchConfig &config);
 
             static std::shared_ptr<BaseOperation>
             generateRetire(const util::UUID &activeCharacter);
 
             static std::vector<std::shared_ptr<BaseOperation>>
-            generateMovementActions(const State &s, const util::UUID &activeCharacter);
+            generateMovementActions(const State &s, const util::UUID &activeCharacter, const MatchConfig &config);
```

Comment:
Warum braucht man für die Movement Actions eine MatchConfig? Bei den Gadgets wird keine verwendet, obwohl dort ja je nach Gadget Reichweiten interessant wären

Reply:
wegen der validate, die in der Funktion aufgerufen wird

Reply:
Aber wenn ich nur Movements erzeuge, kann ich ja auch direkt validateMovement zur Validierung nutzen. Die MatchConfig bräuchte ich ja nur, wenn ich die oberste Verteilerfunktion aufrufen würde.

Reply:
`validateMovement` ist halt private und kann somit nicht direkt aufgerufen werden

Reply:
Spricht was dagegen die public zu machen?

Reply:
Ich glaube es ist sinnvoll die speziellen validations für die generierung zugänglich zu machen. Hier kann man Performance gewinnen, die im Hinblick auf die KI an dieser Stelle relevant sein könnte. Ich würde das hier mal so lassen, da das ein größerer Umbau ist, aber das sollte man im Hinterkopf behalten.

Reply:
Ich habe mal ein issue unter #268 gemacht